Intro-Full/Exercises/kokkoskernels/TeamGemm/Begin/teamgemm.cpp

```diff
@@ -0,0 +1,229 @@
+
+  // Read command line arguments.
+  for (int i = 0; i < argc; i++) {
+    if (strcmp(argv[i], "-A.m") == 0) {
+      A_dims.m = atoi(argv[++i]);
+      printf("  User A.m is %d\n", A_dims.m);
+    }
+    if (strcmp(argv[i], "-A.n") == 0) {
+      A_dims.n = atoi(argv[++i]);
+      printf("  User A.n is %d\n", A_dims.n);
+    }
+    if (strcmp(argv[i], "-B.m") == 0) {
+      B_dims.m = atoi(argv[++i]);
+      printf("  User B.m is %d\n", B_dims.m);
+    }
+    if (strcmp(argv[i], "-B.n") == 0) {
+      B_dims.n = atoi(argv[++i]);
+      printf("  User B.n is %d\n", B_dims.n);
+    }
+    if (strcmp(argv[i], "-alpha") == 0) {
+      user_alpha = atof(argv[++i]);
+      printf("  User alpha %f\n", user_alpha);
+    }
+    if (strcmp(argv[i], "-beta") == 0) {
+      user_beta = atof(argv[++i]);
+      printf("  User beta %f\n", user_beta);
+    }
+
+    else if ((strcmp(argv[i], "-h") == 0) || (strcmp(argv[i], "-help") == 0)) {
+      printf("  TeamGemm Options:\n" );
+      printf("  -A.m <int>:      number of rows in A (default: 32)\n");
+      printf("  -A.n <int>:      number of columns in A (default: 32)\n");
+      printf("  -B.m <int>:      number of rows in B (default: 32)\n");
+      printf("  -B.n <int>:      number of columns in B (default: 32)\n");
+      printf("  -alpha <float>:      value of alpha (default: 1.0)\n");
+      printf("  -beta <float>:       value of beta (default: 2.0)\n");
+      printf("  -help (-h):            print this message\n\n");
+      exit(1);
+    }
+  }
+
+  // Check dimensions of A and B.
+  checkDims(A_dims, B_dims);
+
+  C_dims.m = A_dims.m;
+  C_dims.n = B_dims.n;
+  
+  Kokkos::initialize(argc, argv);
+
+  {
+    // Select View Types
+    // using ScalarType = float;
+    using ScalarType = double;
+    //using Layout = Kokkos::LayoutLeft;
+    using LayoutType = Kokkos::LayoutRight;
+    using DeviceType = Kokkos::Cuda;
+    using ViewType = Kokkos::View<ScalarType***, LayoutType, DeviceType>;
+
+    // Timer products
+    struct timeval begin, end;
+
+    ScalarType alpha = (ScalarType) user_alpha;
+    ScalarType beta = (ScalarType) user_beta;
+    const int N = 1 << 7;
+
+    // Allocate A, B, and C matrices on the device
+    ViewType A("A", N, A_dims.m, A_dims.n);
+    ViewType B("B", N, B_dims.m, B_dims.n);
+    ViewType C("C", N, C_dims.m, C_dims.n);
+
+    // Populate A, B, and C matrices with random numbers
+    using ExecutionSpaceType = DeviceType::execution_space;
+    uint64_t seed = Kokkos::Impl::clock_tic();
+    Kokkos::Random_XorShift64_Pool<ExecutionSpaceType> rand_pool(seed);
+    
+    Kokkos::fill_random(A, rand_pool, Kokkos::rand<Kokkos::Random_XorShift64<ExecutionSpaceType>, ScalarType>::max());
+    Kokkos::fill_random(B, rand_pool, Kokkos::rand<Kokkos::Random_XorShift64<ExecutionSpaceType>, ScalarType>::max());
+    Kokkos::fill_random(C, rand_pool, Kokkos::rand<Kokkos::Random_XorShift64<ExecutionSpaceType>, ScalarType>::max());
+
+    gettimeofday(&begin, NULL);
```

Comment:
In follow-up at some point these timers can be replaced with `Kokkos::Timer` here (and in other exercises as well)